packages/components/src/select-control/types.ts

```diff
@@ -61,50 +61,52 @@ type SelectControlBaseProps = Pick<
 		variant?: 'default' | 'minimal';
 	};
 
-export type SelectControlSingleSelectionProps = SelectControlBaseProps & {
-	/**
-	 * If this property is added, multiple values can be selected. The `value` passed should be an array.
-	 *
-	 * In most cases, it is preferable to use the `FormTokenField` or `CheckboxControl` components instead.
-	 *
-	 * @default false
-	 */
-	multiple?: false;
-	value?: string;
-	/**
-	 * A function that receives the value of the new option that is being selected as input.
-	 *
-	 * If `multiple` is `true`, the value received is an array of the selected value.
-	 * Otherwise, the value received is a single value with the new selected value.
-	 */
-	onChange?: (
-		value: string,
-		extra?: { event?: ChangeEvent< HTMLSelectElement > }
-	) => void;
-};
+export type SelectControlSingleSelectionProps< T extends string = string > =
+	SelectControlBaseProps< T > & {
+		/**
+		 * If this property is added, multiple values can be selected. The `value` passed should be an array.
+		 *
+		 * In most cases, it is preferable to use the `FormTokenField` or `CheckboxControl` components instead.
+		 *
+		 * @default false
+		 */
+		multiple?: false;
+		value?: NoInfer< T >;
```

Comment:
`NoInfer<>` is a [new utility type](https://devblogs.microsoft.com/typescript/announcing-typescript-5-4/#the-noinfer-utility-type) added in TS 5.4, which tells TS not to use this field to infer the generic type. We need this because we want it to infer from the values in the `options` array, not the `value` prop.

This kind of "don't infer" signaling [used to be done](https://www.typescriptlang.org/play/?ts=5.4.5#code/JYWwDg9gTgLgBAKjgQwM5wEoFNkGN4BmUEIcA5FDvmQNwBQdMAnmFnAArFioA8AKnCwAPGFgB2AE3SoYUYGIDmAPjgBeOAG86cOADdkAGwCuWAFxwBAMjgy5i+johgYwCGNTmtOnfuNmLDnAAvgDaALr0QfR0BEZi+K5icADCJJBi4jD8giLiUjay8soAFGBcHhzl-EoAlJracJQwRlBJYkYGBpEMAPQ9cACiUMRQ6EZgcDAQegCsAHQAzItwAEZG8GIQ8MgEolCzcwAsczN0PKngbpl6hiaqZAQQEGRwTi5uqKoaIRq+JuZkFbIKBkIJhIJwHpKejnNJXMTwP5Ye5AkGvZyJT7fX63fyA4Gg8GQ6FAA) with `T & string`, but does not work anymore in TS 5.4+.

Reply:
Can it be demonstrated when the `NoInfer` can be useful? A new test case that fails without it? I don't understand very well what it does.

Reply:
Sure, Without `NoInfer` Typescript will infer the generic type from the value prop as well as the options prop.

What this means is it won't detect that your value is not one of the provided options, and therefore won't provide an error for it.

I've created a [simple TS Playground example](https://www.typescriptlang.org/play/?#code/GYVwdgxgLglg9mABFApgZygdRlAFnEKAOTgEkxgUAnAHgBVEUAPVMAEzUQypjAHMAfAAoAbgEMANiBQAuRHQA0icVPRy6AbQC6AShkMA3oioooIKkhXTEAXwBQEBBkQEoAB0KIAvMnRYc+IQk5JRUQgBE3Lx84Uoa4SowbLGI4XBusE7hugDcdgD0+YjFiAB6APx2dqCQmUioGNh4wRTU9IwsKOycUfzCVrKILaH0AkoDaOraeobGpuaWktb2jmDOrh5QAEzevo0Bw9QRvTFxCZJJKWkZ8GvZOnmFJWWVQA) to illustrate the difference for you. Hope that helps 🙂

Reply:
Thanks, this is nice 🙂 I didn't realize that passing something as `value` can expand the `T` type to also include the type of `value`. But we want the type to be checked against `options` instead.

Reply:
Nice to see a clever usage of `NoInfer` 👍 I had read about it, but struggled to think of a good use case for it.

Reply:
Also found a good article about NoInfer: https://www.totaltypescript.com/noinfer